src/main.rs

```diff
@@ -2,6 +2,8 @@ use std::env;
 use std::fs;
 use std::path::{Path, PathBuf};
 use std::process::exit;
+use std::env::consts::ARCH;
```

Comment:
Unclear from docs if this is resolved at runtime variable or compile time. Do you know for certain?

Reply:
> Unclear from docs if this is resolved at runtime variable or compile time. Do you know for certain?

its compile time... but given it is a compiled language, I am not sure compile time or runtime would make much difference:

https://github.com/rust-lang/rust/blob/6d1fc53cf4f61b0dce5fc84cebae3b9fb8a45436/library/std/src/env.rs#L906C28-L906C31

which relies on the env macro, which itself expands at compile time https://doc.rust-lang.org/std/macro.env.html

Reply:
I think some systems might have binfmt_misc configured. So a runtime mechanism will probably be more reliable in those cases. So if it's not too complicated, a runtime mechanism is probably better here

Reply:
> I think some systems might have binfmt_misc configured. So a runtime mechanism will probably be more reliable in those cases. So if it's not too complicated, a runtime mechanism is probably better here

I am trying to understand what that would solve. Because the vmtest binary is compiled for each specific arch, the value of ARCH should be set to the platform vmtest was compiled for.

I suppose using runtime value (as whatever would be extracted from `uname -m`) would only help if one runs vmtest from a foreign architecture? Correct?

Anyway, I will close this draft for now. My main goad was to showcase the goal.

I will split this out into its own PRs, the 2 commits here, as much as they complement each others, are independent. It will be better to have this discussion there.

Reply:
> > nk some systems might have binfmt_misc configured. So a runtime mechanism will probably be more reliable in those cases. So if it's not too complicated, a runtime mechanism is probably better here
> 
> I am trying to understand what that would solve. Because the vmtest binary is compiled for each specific arch, the value of ARCH should be set to the platform vmtest was compiled for.
> 
> I suppose using runtime value (as whatever would be extracted from `uname -m`) would only help if one runs vmtest from a foreign architecture? Correct?

Yeah. binfmt_misc can transparently run a foreign binary by invoking qemu (or similar). Sorta like how M1 macs can run x86 programs transparently. It's a niche error case for sure, but just point it out

Reply:
> Yeah. binfmt_misc can transparently run a foreign binary by invoking qemu (or similar). Sorta like how M1 macs can run x86 programs transparently. It's a niche error case for sure, but just pointing it out

yeah, it feels like this may only happen because on invokes a build for a foreign arch instead of the one for its current arch.
I feel this would be a problem if vmtest could not be compiled for a given target arch. Or maybe for people using the github action?
Looks like https://doc.rust-lang.org/std/env/consts/constant.ARCH.html aligns with what you would get from `uname -m` so maybe this can be used instead.